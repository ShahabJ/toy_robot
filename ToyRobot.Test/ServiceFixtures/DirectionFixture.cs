using System;
using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using ToyRobot.Library.Interface;
using ToyRobot.Library.Model;
using ToyRobot.Library.Service;
using Xunit;

namespace ToyRobot.Test.ServiceFixtures
{
    [UseReporter(typeof(DiffReporter))]
    public class DirectionFixture
    {
        private Position position;
        private readonly int x = 1;
        private readonly int y = 1;

        [Fact]
        public void GenerateReport()
        {
            var report = new List<string>();
            var directions = new List<IDirection>()
                {
                    new NorthDirection(),
                    new EastDirection(),
                    new WestDirection(),
                    new SouthDirection()
                };

            foreach (var direction in directions)
            {
                position = new Position(x, y, Enum.Parse<DirectionEnum>(direction.GetName()));
                report.Add($"Current Position:{position},New position after Move:{direction.Move((Position)position.Clone())}");
                report.Add($"Current Position:{position},New position after TurnLeft:{direction.TurnLeft((Position)position.Clone())}");
                report.Add($"Current Position:{position},New position after TurnRigh:{direction.TurnRight((Position)position.Clone())}");
            }
            Approvals.VerifyAll(report, "");
        }
    }
}