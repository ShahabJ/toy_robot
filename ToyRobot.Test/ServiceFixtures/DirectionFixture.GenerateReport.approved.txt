[0] = Current Position:1,1,NORTH,New position after Move:1,2,NORTH
[1] = Current Position:1,1,NORTH,New position after TurnLeft:1,1,WEST
[2] = Current Position:1,1,NORTH,New position after TurnRigh:1,1,EAST
[3] = Current Position:1,1,EAST,New position after Move:2,1,EAST
[4] = Current Position:1,1,EAST,New position after TurnLeft:1,1,NORTH
[5] = Current Position:1,1,EAST,New position after TurnRigh:1,1,SOUTH
[6] = Current Position:1,1,WEST,New position after Move:0,1,WEST
[7] = Current Position:1,1,WEST,New position after TurnLeft:1,1,SOUTH
[8] = Current Position:1,1,WEST,New position after TurnRigh:1,1,NORTH
[9] = Current Position:1,1,SOUTH,New position after Move:1,0,SOUTH
[10] = Current Position:1,1,SOUTH,New position after TurnLeft:1,1,EAST
[11] = Current Position:1,1,SOUTH,New position after TurnRigh:1,1,WEST
